alws/crud/build_node.py

```diff
@@ -98,62 +107,107 @@ async def check_build_task_is_finished(db: Session, task_id: int) -> bool:
     return BuildTaskStatus.is_finished(build_task.status)
 
 
+async def __get_rpm_package_info(pulp_client: PulpClient, rpm_href: str) \
+        -> (str, dict):
+    info = await pulp_client.get_rpm_package(rpm_href)
+    return rpm_href, info
+
+
 async def __process_rpms(pulp_client: PulpClient, task_id: int, task_arch: str,
                          task_artifacts: list, repositories: list,
                          built_srpm_url: str = None, module_index=None):
+
+    def get_repo(repo_arch, is_debug):
+        return next(
+            build_repo for build_repo in repositories
+            if build_repo.arch == repo_arch
+            and build_repo.type == 'rpm'
+            and build_repo.debug == is_debug
+        )
+
     rpms = []
+    arch_repo = get_repo(task_arch, False)
+    debug_repo = get_repo(task_arch, True)
+    src_repo = get_repo('src', False)
+    arch_packages_tasks = []
+    src_packages_tasks = []
+    debug_packages_tasks = []
     for artifact in task_artifacts:
-        arch = task_arch
-        if artifact.arch == 'src':
-            arch = artifact.arch
-        if arch == 'src' and built_srpm_url is not None:
-            continue
-        repo = next(
-            build_repo for build_repo in repositories
-            if build_repo.arch == arch
-            and build_repo.type == artifact.type
-            and build_repo.debug == artifact.is_debuginfo
+        if artifact.arch == 'src' and built_srpm_url is None:
+            src_packages_tasks.append(create_entity(pulp_client, artifact))
+        if artifact.is_debuginfo:
+            debug_packages_tasks.append(create_entity(pulp_client, artifact))
+        else:
+            arch_packages_tasks.append(create_entity(pulp_client, artifact))
+
+    processed_packages = []
+    for tasks, repo in (
+            (src_packages_tasks, src_repo), (arch_packages_tasks, arch_repo),
+            (debug_packages_tasks, debug_repo)):
+        if tasks:
+            try:
+                results = await asyncio.gather(*tasks)
```

Comment:
why not all of them together?

Reply:
Each set of artifacts needs to be converted and then added to the corresponding repo. Tried to not overcomplicate logic here